### ISSUE TITLE
Invalid access token leads to "Connectivity to the server has been lost" with no way to gracefully recover - need to manually logout and re-login

### ISSUE THREAD

**Issue by toby63** [element-hq/element-web#23063]
### Steps to reproduce

1. Lost connection to server, probably because of a internet connection issue from Server side.
2. Checked that the Server is back up.
3. Tried to send messages etc., but Client still insists that the Server is not available.
4. Searched in the server logs, found error ` SynapseError: 401 - Invalid Authorization header.`
5. Searched the internet and found this: https://github.com/vector-im/element-web/issues/14529#issuecomment-879849922 
6. After Logout and Re-login everything was working normal, just as suggested by the user.

#### What did you expect?

Client should be able to reconnect to a server by itself without a logout being necessary.
Client should at least show a useful error message.

#### What happened instead?

Client insisted that a connection is not possible without any explanation.

### Operating system

Manjaro Linux

### Application version

Element v1.11.2

By the way, this problem also occurs with the Android App.

### How did you install the app?

Manjaro Repos

### Homeserver

Synapse 1.64.0

### Will you send logs?

No

**Comment by MadLittleMods** [comment]
It sounds like your access token is no longer valid so you can't authenticate and communicate with the homeserver anymore. Your token could have been invalidated in a number of different ways. Is there anything special about your homeserver setup (SAML, SSO, CAS)?

> The client should handle this situation automatically either by refreshing the token, or by showing login form again (worse, but still less frustrating than just permanent disconnect).
>
> *-- https://github.com/vector-im/element-web/issues/14529#issuecomment-879849922*

As the other user suggested, when the client sees a `401 - Invalid Authorization header` response from the homeserver, it should ask you to sign in again.

Maybe this means completely signing you out but it seems a bit more friendly if we just update the [`RoomStatusBar`](https://github.com/matrix-org/matrix-react-sdk/blob/8db7766a4012f6ffa0707795698513c585ddaee1/src/components/structures/RoomStatusBar.tsx) to take this situation into account and explain that you need to sign in again since your token is no longer valid.

**Comment by toby63** [comment]
> Your token could have been invalidated in a number of different ways. Is there anything special about your homeserver setup ([SAML, SSO, CAS](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html))?

Not that I know of.
I don't use any Single-Sign-On Service.

**Comment by MadLittleMods** [comment]
@toby63 On your homeserver, do you have [`session_lifetime`](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#session_lifetime) configured in your `homeserver.yaml`?

It looks like the SSO stuff doesn't matter anyway as it doesn't destroy your session at all atm, https://github.com/matrix-org/synapse/issues/4158

**Comment by toby63** [comment]
> @toby63 On your homeserver, do you have [`session_lifetime`](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#session_lifetime), [`refreshable_access_token_lifetime`](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#refresh_access_token_lifetime), [`refresh_token_lifetime`](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#refresh_token_lifetime-24h), or [`nonrefreshable_access_token_lifetime`](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#nonrefreshable_access_token_lifetime) configured in your `homeserver.yaml`?

No, everything is still commented out.
I can maybe take a look at the config tomorrow and cut out all private details and then post it here or send it to you.
But everything is really basic, it's just a little server for family and friends, nothing special is added to it etc.

**Comment by MadLittleMods** [comment]
@toby63 From another device, is it possible that you cleaned up your devices/sessions yourself? **Settings** -> **Security & Privacy** -> **Where you're signed in** -> and signed out some devices

**Comment by toby63** [comment]
> @toby63 From another device, is it possible that you cleaned up your devices/sessions yourself? **Settings** -> **Security & Privacy** -> **Where you're signed in** -> and signed out some devices

No.
This problem occured with multiple accounts, also with other users on my server.

**Comment by raingloom** [comment]
I'm probably running into the same error.  My friends who are on the same server have no issues connecting, but I have been unable to connect for several days.  I guess I could log out and log back in, but I have some queued messages I don't want to lose.

**Comment by MadLittleMods** [comment]
@raingloom @toby63 Submit some debug logs from your Element client when this happens. There might be some more info to go from.

**Comment by Polve** [comment]
I have one PC where everytime I quit the desktop client and reopen it I get this error, rendering the program unusable.

So I need to logout, relogin e re-verify the new login every day.

Where may I check some logging to try to understand a bit better what is the problem?

**Comment by MadLittleMods** [comment]
> Where may I check some logging to try to understand a bit better what is the problem?

You can use `/rageshake` to quickly jump to the **Submit debug logs** dialog where you can download and submit the logs.

The same dialog can be reached by going to the profile menu in the top-left -> **Settings** -> **Help & About** -> **Submit debug logs**

**Comment by alinelena** [comment]
i have exactly the same issue... crazy enough on a clean install... my android and two other linux computers do not have this issue.

**Comment by raymand211092** [comment]
Related issues https://github.com/SchildiChat/schildichat-desktop/issues/170

**Comment by johnjaylward** [comment]
I'm having the same issue, and it seems to be related to some kind of database corruption in the client as I see this lin in my logs for when the client starts up:

> "Error opening the database: DatabaseUnlockError(\"Invalid passphrase\")"

key log line:

```
2023-11-09T18:08:30.938Z D StorageManager: Checking storage consistency
2023-11-09T18:08:30.938Z D StorageManager: Local storage supported? true
2023-11-09T18:08:30.938Z D StorageManager: IndexedDB supported? true
2023-11-09T18:08:30.938Z D StorageManager: Local storage contains data? true
2023-11-09T18:08:30.938Z D StorageManager: Crypto initialised? true
2023-11-09T18:08:30.940Z D StorageManager: Sync store using IndexedDB contains data? true
2023-11-09T18:08:30.940Z D StorageManager: Crypto store using IndexedDB contains data? true
2023-11-09T18:08:30.940Z D StorageManager: Storage consistency checks passed
2023-11-09T18:08:30.968Z E Expected a pickle key, but none provided.  Encryption may not work.
2023-11-09T18:08:30.968Z D Session persisted for @redacted:matrix.redacted.server
2023-11-09T18:08:30.968Z D Lifecycle: Starting MatrixClient
2023-11-09T18:08:30.972Z D FetchHttpApi: --> GET https://matrix.redacted.server/_matrix/client/v3/thirdparty/protocols
2023-11-09T18:08:30.978Z D StorageManager: Persistent? true
2023-11-09T18:08:30.979Z D Switching to room id !qfhFJYaXaUcoggTKJK:matrix.redacted.server at event undefined
2023-11-09T18:08:31.180Z D EventIndex: Error initializing the event index {"message":"Error opening the database: DatabaseUnlockError(\"Invalid passphrase\")"}
2023-11-09T18:08:31.180Z D IndexedDBStore.startup: connecting to backend
2023-11-09T18:08:31.180Z D MatrixClientPeg: waiting for MatrixClient store to initialise
2023-11-09T18:08:31.197Z D IndexedDB worker is ready
```

Once loading the session failed, the client continues to try to process the connections. However, it appears that this "Invalid passphrase" error is not being properly handled. Either the user should be asked to re-enter the passphrase, or the client should explain the error and ask the user to re-auth, then restart the session.

**Comment by johnjaylward** [comment]
My issue appears to be KDE/Gnome keystore issue. After logging in, I stepped away from the keyboard and the keystore prompted me for my passphrase, but I didn't enter it until I returned some 30+minutes later. Meanwhile, Element-Desktop started up (as I have it set to auto-start on login) and the keystore was not unlocked yet.

Logging out of my screen and back in the error did not repeat itself.
